### ISSUE TITLE
 Keras backend categorical_crossentropy behaving differently from default one

### ISSUE THREAD

**Issue by shashankg7** [keras-team/keras#7973]
I am trying to implement weighted categorical_crossentropy loss in keras. The loss I am trying to implement is:

            alpha * categorical_crossentropy(y_true, y_pred)

where alpha is the weight. If I pass the loss function as following:

    def loss(y_true, y_pred):
         return alpha * K.categorical_crossentropy(y_true, y_pred)

and pass it in the model compile phase, the loss drops to a very small value (in range of 1e-8) for some epochs and final acc. is 0. 

I tried removing the alpha from the custom loss function and only return K.categorical_crossentropy, still same issue.

Why is it behaving differently then simple passing the string 'categorical_crossentropy' in the model compile phase? Behind the scenes, it must be calling the same function, I suppose.

**Comment by julienr** [comment]
It indeed calls the same function [behind the scene](https://github.com/fchollet/keras/blob/master/keras/losses.py#L98). 

Can you provide a minimal example to reproduce the problem ?

**Comment by MostafaAlaa2016** [comment]
@shashankg7 Did you solve it? I have the same issue

I tried  both methods and it behaves differently than if I pass the original function

```
def w_categorical_crossentropy(y_true, y_pred,classes_dest,encoder,sum_of_classes_denesity ):
    """ # Custom loss function with costs """
    return K.categorical_crossentropy(y_pred, y_true)
```


```

def w_categorical_crossentropy2(y_true, y_pred):
    """ # Custom loss function with costs """
    return categorical_crossentropy(y_pred, y_true) 
```

If I passed loss='categorical_crossentropy' it working fine but if I passed any of the above it stuck at 24% and not increasing.

I am calling the function using two ways of partial functions
```

ncce = partial(w_categorical_crossentropy2)
ncce.__name__ ='w_categorical_crossentropy2'
```

OR
```

w_categorical_crossentropy_Pfun = wrapped_partial(w_categorical_crossentropy, classes_dest = classes_dest,encoder = classes_encoder,sum_of_classes_denesity = sum_of_classes_denesity)

model.compile(optimizer = 'adam', 
                      #loss=w_categorical_crossentropy_Pfun, this or the below line both not working
                      loss=ncce,
                      metrics = ['accuracy'])
```

**Comment by MostafaAlaa2016** [comment]
Some more information about my model 
Model Arch below
`_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
lstm_5 (LSTM)                (None, 84, 50)            11800     
_________________________________________________________________
lstm_6 (LSTM)                (None, 84, 50)            20200     
_________________________________________________________________
lstm_7 (LSTM)                (None, 84, 50)            20200     
_________________________________________________________________
lstm_8 (LSTM)                (None, 50)                20200     
_________________________________________________________________
dense_2 (Dense)              (None, 12)                612       
=================================================================
Total params: 73,012
Trainable params: 73,012
Non-trainable params: 0
_________________________________________________________________
None`

**Comment by MostafaAlaa2016** [comment]
@julienr Do you have any Idea how to solve this issue?

**Comment by julienr** [comment]
@MostafaAlaa2016 in your ` w_categorical_crossentropy` implementation, you do
 `return K.categorical_crossentropy(y_pred, y_true)`
but looking at [keras' code](https://github.com/keras-team/keras/blob/master/keras/losses.py#L68), this should be
`return K.categorical_crossentropy(y_true, y_pred)`
(swapped `y_true` and `y_pred`).

Does that solve your issue ?

**Comment by MostafaAlaa2016** [comment]
@julienr yes, thanks for your help 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>
